# [LAB-7] 2. 가설검정의 이해 - 연습문제

## #01. 준비작업

### 1. 라이브러리 참조

In [1]:
from jussam import load_data
from helpers import my_stats

📦 연세대학교 주영아 교수가 제작한 라이브러리를 사용중입니다.
📧 Email(1): j.purplerose@yonsei.ac.kr
📧 Email(2): j.purplerose@gmail.com
📝 Website: https://juyounga.kr/


### 2. 데이터 가져오기

In [2]:
origin = load_data("icecream")
origin.head()

📚 기온과 아이스크림 판매량을 기록한 가상의 데이터 (메타데이터,인덱스 없음)


,기온,판매량
0,23,431
1,36,593
2,30,512
3,25,474
4,26,476


## #02. 가설검정의 가정 확인

In [3]:
my_stats.test_assumptions(origin, columns=["기온", "판매량"])

,test,statistic,p-value,result
field,,,,
기온,normaltest,2.745,0.253,True
판매량,normaltest,0.715,0.700,True
Bartlett,equal_var,44.148,0.000,False


## 문제 풀이

### 문제 1

유의수준 0.05에서 정규성 검정 결과를 해석하고, 기온과 판매량이 정규성 가정을 만족하는지 답하시오.

### 정답 및 해설

- **정답**: 두 변수 모두 정규성을 만족한다.
- **해설**: 정규성 검정의 귀무가설(H₀)은 "데이터가 정규분포를 따른다"이다. 기온(p=0.253)과 판매량(p=0.700) 모두 p-value가 0.05보다 크므로 귀무가설을 기각할 수 없다. 따라서 정규성 가정을 충족하며 result는 True이다.

### 문제 2

정규성 검정 결과에 따라 기온과 판매량의 등분산성을 검정하기 위해 사용되는 검정 방법은 무엇인가?

- **정답**: Barttlet 검정
- **해설**: 모든 변수가 정규성을 만족할 경우 Barttlet 검정을 사용하고 그렇지 않을 경우 Lavene 검정을 사용한다.

### 문제 3

두 변수가 등분산 가정을 만족하는지 판단하시오.

- **정답**: 등분산 가정을 만족하지 않는다
- **해설**: Bartlett 검정의 H₀는 "두 집단의 분산이 같다"이다. p-value가 0.000으로 0.05보다 작아 귀무가설을 기각하므로 분산이 서로 다르다고 본다. result는 False이다.